[CI] Add [optional] monitoring component

lukas-vlcek opened the pull request:
Adding an option to deploy `openshift-monitoring` component into `hack/get_machine_run_tests.sh`.
We will be able to benefit from this in 4.0 and later after Prometheus integration is better supported in origin parts.

I assume this PR will need couple of review cycles before it is ready.

richm commented on hack/get-machine-run-tests.sh:
```diff
@@ -734,6 +734,26 @@ if [ -n "${PRESERVE:-}" ] ; then
 fi
 
+#      title: "install origin-monitoring"
+#      repository: "openshift-ansible"
```
I suppose, but it doesn't really matter

ewolinetz replied:
I vote for `aos-cd-jobs` for the sake of consistency

lukas-vlcek replied:
In the beginning I do `cd $OS_A_C_J_DIR` but in the end when running the playbook I use path `$OS_O_A_DIR/playbooks/openshift-monitoring/config.yml`. So, I am not sure. I guess most of the stuff is used from `$OS_A_C_J_DIR`. In that case @ewolinetz 's suggestion would make more sense.

richm replied:
you have to `cd $OS_A_C_J_DIR` because the inventory is in `$OS_A_C_J_DIR/sjb/inventory`

richm commented on hack/get-machine-run-tests.sh:
```diff
@@ -739,11 +739,14 @@ fi
+  --connection local \
+  --inventory sjb/inventory/ \
+  -e openshift_deployment_type=origin  \
+  $OS_O_A_DIR/playbooks/openshift-monitoring/config.yml \
```
For consistency, the playbook should be the last argument on the command line, after any `-e` arguments

richm commented:
/test logging
remote syslog flake

ewolinetz commented:
/hold

(this will stop the bot from merging)
to remove the hold from the bot add the same command with `cancel`

lukas-vlcek commented:
Updated and squashed. What do you think we should add before we can merge it into master? By default the monitoring component is not installed. Only when `$USE_MONITORING=true`. As of writing this component is not integrated with logging at all, but we will see more integration going forward - Prometheus, Alerts & Grafana dashboard(s).

richm commented:
> What do you think we should add before we can merge it into master?

Nothing.  I think it is fine as it is.  It solves the problem of installing monitoring.

> but we will see more integration going forward - Prometheus, Alerts & Grafana dashboard(s).

. . . and we can add those as subsequent PRs

richm commented:
/lgtm

ewolinetz commented:
/hold cancel
